src/components/Translation.tsx

```diff
@@ -20,14 +19,41 @@ const transform = {
 // Renders the translation string for the given translation key `id`. It
 // fallback to English if it doesn't find the given key in the current language
 const Translation = ({ id, options }: Props) => {
-  // const { t } = useTranslation()
+  const [requiredNamespaces, setRequiredNamespaces] = useState<string[]>([])
+  const { asPath } = useRouter()
 
-  // const translatedText = t(id, options)
+  useEffect(() => {
+    if (asPath.startsWith("/community")) {
+      setRequiredNamespaces([...requiredNamespaces, "page-community"])
+    }
+
+    if (asPath.startsWith("/energy-consumption")) {
+      setRequiredNamespaces([
+        ...requiredNamespaces,
+        "page-about",
+        "page-what-is-ethereum",
+      ])
+    }
+
+    if (asPath.startsWith("/history")) {
+      setRequiredNamespaces([...requiredNamespaces, "page-history"])
+    }
+
+    if (asPath.startsWith("/nft")) {
+      setRequiredNamespaces([...requiredNamespaces, "learn-quizzes"])
+    }
+
+    if (asPath.startsWith("/staking")) {
+      setRequiredNamespaces([...requiredNamespaces, "page-staking"])
+    }
+  }, [requiredNamespaces])
+
+  const { t } = useTranslation(requiredNamespaces)
+  const translatedText = t(id, options)
 
   // Use `htmr` to parse html content in the translation text
   // @ts-ignore
-  // return <>{htmr(translatedText, { transform })}</> TODO: add this return later
-  return null
+  return <>{htmr(translatedText, { transform })}</>
```

Comment:
Is this what allows us to use things like `"some-string": "Hello <strong>bolded world</strong>"` in our JSON strings?

Reply:
Yes, was previously on the codebase. I noticed that for some cases (bold text) is not rendering spaces between strings, we need QA to debug these cases.